src/payment/strategies/checkoutcom/checkoutcom-googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,128 @@
+import { CheckoutActionCreator, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import { ContentType } from '../../../common/http-request/';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import Payment from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategyActionCreator from '../../payment-strategy-action-creator';
+import { GooglePayPaymentProcessor, GooglePayPaymentStrategy } from '../googlepay';
+
+import { CheckoutcomGooglePayToken, CheckoutcomToken } from './checkout';
+
+export default class CheckoutcomGooglepayPaymentStrategy extends GooglePayPaymentStrategy {
+    TEST_URL: string = 'https://api.sandbox.checkout.com/tokens';
+    LIVE_URL: string = 'https://api.checkout.com/tokens';
+
+    constructor(
+        _store: CheckoutStore,
+        _checkoutActionCreator: CheckoutActionCreator,
+        _paymentMethodActionCreator: PaymentMethodActionCreator,
+        _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        _paymentActionCreator: PaymentActionCreator,
+        _orderActionCreator: OrderActionCreator,
+        _googlePayPaymentProcessor: GooglePayPaymentProcessor
+    ) {
+        super(_store,
+            _checkoutActionCreator,
+            _paymentMethodActionCreator,
+            _paymentStrategyActionCreator,
+            _paymentActionCreator,
+            _orderActionCreator,
+            _googlePayPaymentProcessor);
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this.getGooglePayOptions()) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
+        }
+
+        const { payment: { methodId = '' } = {} } = payload;
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this.getStore().getState();
+
+        const testMode: boolean | undefined = getPaymentMethodOrThrow(methodId).config.testMode;
```

Comment:
can this just be boolean for simplicity?
```js
const testMode: boolean = !!getPaymentMethodOrThrow(methodId).config.testMode;
```

Reply:
changed as suggested